main/src/io/github/iltotore/iron/RuntimeConstraint.scala

```diff
@@ -0,0 +1,11 @@
+package io.github.iltotore.iron
+
+type RuntimeConstraint[T] = T match
+  case IronType[a, c] => RuntimeConstraint.Impl[a, c, T]
```

Comment:
@Iltotore do you see a reason why this would be a bad idea?

Reply:
It will be inconsistent with opaque types. This bug will be fixed in the next versions of Scala.

Reply:
Okay after trying a few things out, this really doesn't play well with opaques so I reverted this change